src/NumericalAlgorithms/Spectral/Basis.hpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Declares functions for retrieving Legendre-Gauss-Lobatto related matrices
+
+#pragma once
+
+#include <cstddef>
+
+class DataVector;
+class Matrix;
+
+/*!
+ * \ingroup SpectralGroup
+ * \brief Basis functions, derivative matrices, etc. for spectral-type methods
+ */
+namespace Basis {
+
+/*!
+ * \brief A particular set of basis functions
+ */
+enum class Polynomials {
```

Comment:
- How about `Basis::Functions`, or `Basis::Orthogonal`? `Basis::Basis1D` has this inelegant duplication, and there is no `2D` or else, so it's a bit redundant perhaps?
- Or we could namespace it to read `Basis::Legendre::collocation_points<Quadrature::Gauss>(num_pts)` at the call site?
- Or we move it to the namespace `Spectral`, with enum `Basis`, so that it reads `Spectral::Basis::Legendre`, and at the call site:

  ```c++
  Spectral::collocation_points<
    Spectral::Basis::Legendre,
    Spectral::Quadrature::Gauss
  >(num_pts)
  ```

Reply:
okay, the last one sounds good